### PR TITLE
[CORL-3186]: Update docs packages and add lunrjs search

### PR DESCRIPTION
**Pull request by kabeaty**
## What does this PR do?

These are changes that are looking to update Coral's docs to the latest version of Docusaurus, plus switches out algolia search in favor of using `lunrjs` for search functionality. It also updates how we are generating the API docs, now using `@graphql-markdown/docusaurus`.

Everything should be basically the same. The API docs have a slightly different layout, with `Operations` and `Types` categories now.

## These changes will impact:

- [ ] commenters
- [ ] moderators
- [ ] admins
- [ ] developers

## What changes to the GraphQL/Database Schema does this PR introduce?

none

## Does this PR introduce any new environment variables or feature flags?

no

## If any indexes were added, were they added to `INDEXES.md`?

n/a

## How do I test this PR?

You can run `pnpm run build` and `pnpm run serve` in `docs` to see the docs locally.

### CODE REVIEW

File: docs/.eslintignore [hunk 1]
```diff
@@ -2,4 +2,5 @@
 *.graphql.ts
 **/__generated__/**
 docs/build
+build
```
Comment (nick-funk) [inline docs/.eslintignore:5]: ❤️ 🙉

File: docs/README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Website
 
-This website is built using [Docusaurus 2](https://docusaurus.io/), a modern static website generator.
+This website is built using [Docusaurus](https://docusaurus.io/), a modern static website generator.
```
Comment (nick-funk) [inline docs/README.md:3]: lol, this makes me wonder if there was a major distinction between v1 and v2 for Docusaurus or something

File: docs/docs/sso.md [hunk 3]
```diff
@@ -108,7 +108,7 @@ as follows:
   `requestAccountDeletion` does.
 - `requestUserCommentsDownload` lets you retrieve a given account's comments download. This mutation will provide you with a `archiveURL` that can be used to download a ZIP file containing the user's comment export.
 
-If you're unsure on how to call GraphQL API's, refer to the section here on [Making your first GraphQL request](/api/schema#making-your-first-request).
+If you're unsure on how to call GraphQL API's, refer to the section here on [Making your first GraphQL request](/api/schema#quick-start).
```
Comment (nick-funk) [inline docs/docs/sso.md:111]: `quick-start` is a much nicer and more common term, +1

File: docs/package.json [hunk 4]
```diff
@@ -39,15 +55,12 @@
...
+      "last 5 safari version"
     ]
   },
-  "devDependencies": {
-    "@docusaurus/types": "^2.0.0-beta.1",
-    "@typescript-eslint/eslint-plugin": "^5.62.0",
-    "@typescript-eslint/eslint-plugin-tslint": "^5.62.0",
-    "@typescript-eslint/parser": "^5.62.0"
+  "engines": {
+    "node": ">=18.0"
```
Comment (nick-funk) [inline docs/package.json:64]: heads up, can't remember if we host this via Netlify or not, but you'll want to make sure its runner/deploy is updated to this Node version if you want to specify it like this

it's quite possible it's running a deprecated v14/16 version in its config

File: docs/static/_redirects [hunk 5]
```diff
@@ -1 +1 @@
-/migrate-from-talk /migrating-from-4
+/migrate-from-talk /migrating-from-4
```
Comment (nick-funk) [inline docs/static/_redirects:1]: +1, nice update

**Review by nick-funk — APPROVED** [review]
Looks good, be sure to check my comment about Netlify for this to deploy out cleanly!

File: docs/README.md (hunk 2 above)
Comment (kabeaty) [reply docs/README.md:3]: lol, not sure, this was part of Docusaurus updates when I generated a fresh docs to replace the old version

File: docs/package.json (hunk 4 above)
Comment (kabeaty) [reply docs/package.json:64]: Confirmed that we're good here--I checked the latest deploy log and it is using node 18 already from `.nvmrc` and deployed a preview just fine

**Pull request merged**